docs(button): addded danger variants (#2026)
carbon-design-system/carbon-website: src/pages/components/button/usage.mdx

| _Primary_   | For the principal call to action on the page. Primary buttons should only appear once per screen (not including the application header or in a modal dialog). |
| _Secondary_ | For secondary actions on each page; these can only be used  in conjunction with a primary button.                                                             |
| _Tertiary_  | For less prominent actions; tertiary buttons can be used in isolation  or paired with a primary button when there are multiple calls to action.               |

> | _Danger_    | For actions that could have destructive effects on the user’s data  (delete, remove, etc.). Danger button has three styles: primary, tertiary, and ghost.     |

Reviewer: ```suggestion
| _Danger_    | For actions that could have destructive effects on the user’s data  (for example, delete or remove). Danger button has three styles: primary, tertiary, and ghost.     |
```

</Column>
</Row>

### Danger button variations

> The danger button has three different styles: primary, tertiary and ghost.

Reviewer: ```suggestion
The danger button has three different styles: primary, tertiary, and ghost.
```

Determining which danger button style to use will depend on the level of
emphasis you want to give to the danger action. Destructive actions that are

> considering a required or primary step in a workflow should use the primary

Reviewer: ```suggestion
considered a required or primary step in a workflow should use the primary
```

danger button style. However, if a destructive action is just one of several

> actions a user could choose then a lower emphasis style like the tertiary danger

Reviewer: ```suggestion
actions a user could choose from, then a lower emphasis style like the tertiary danger
```

button or the ghost danger button may be more appropriate.

<Row>
<Column colLg={8}>

![Danger button variations](images/button_usage_31.png)

> <Caption>Examples of the primary, tertiary and ghost danger buttons</Caption>

Reviewer: ```suggestion
<Caption>Examples of the primary, tertiary, and ghost danger buttons</Caption>
```
